pkg/cstor/volume/v1alpha1/cstorvolume.go

```diff
@@ -86,16 +108,53 @@ type Predicate func(*CStorVolume) bool
 
 // IsHealthy returns true if the CVR is in
 // healthy state
-func (p *CStorVolume) IsHealthy() bool {
-	return p.object.Status.Phase == "Healthy"
+func (c *CStorVolume) IsHealthy() bool {
+	return c.object.Status.Phase == "Healthy"
 }
 
 // IsHealthy is a predicate to filter out cstorvolumes
 // which is healthy
 func IsHealthy() Predicate {
-	return func(p *CStorVolume) bool {
-		return p.IsHealthy()
+	return func(c *CStorVolume) bool {
+		return c.IsHealthy()
+	}
+}
+
+// IsResizePending return true if resize is in progress
+func (c *CStorVolume) IsResizePending() bool {
+	curCapacity := c.object.Status.Capacity
+	desiredCapacity := c.object.Spec.Capacity
+	if curCapacity.IsZero() {
```

Comment:
why for 'IsZero' returning 'false'?

Reply:
Basic intention is during resize if controller is restarted then target will be created with updated size then to know whether we resize is in process we are not updating the status caoacity.

Reply:
When resize is success then we need to remove the conditions update the conditions if they are present.

Reply:
By default curCapacity will have 0